web/src/Api/Endpoint.elm

```diff
@@ -48,10 +50,46 @@ type StudentEndpoint
     = StudentEndpoint
 
 
+type StudentLogoutEndpoint
+    = StudentLogoutEndpoint
+
+
+type StudentResearchConsentEndpoint
+    = StudentResearchConsentEndpoint
+
+
+type StudentUsernameValidEndpoint
+    = StudentUsernameValidEndpoint
+
+
+studentValidUsernameEndpoint : Api.Config.Config -> Endpoint StudentUsernameValidEndpoint
```

Comment:
Could we call these something like `studentValidUsername` and agree to always call them off of `Endpoint` like `Endpoint.studentValidUsername`?

Reply:
Sure, I wouldn't mind that.